File: typescript/src/api/resources/traceImports/client/requests/CreateTraceImportRequest.ts

import * as Sikaru from "../../../../index.js";

/**
 * @example
 *     {
 *         connectionId: "connectionId",
 *         converterVersion: "converterVersion",
 *         dataset: "dataset",
 *         externalProjectId: "externalProjectId",
 *         mode: "historical",
 *         provider: "langsmith",
 *         scope: {},
 *         sourceInstance: "sourceInstance"
 *     }
 */
export interface CreateTraceImportRequest {
    connectionId: string;
    converterVersion: string;
    dataset: string;
    externalProjectId: string;
    grouping?: Sikaru.TraceImportGroupingRequest;
    mode: CreateTraceImportRequest.Mode;
    options?: Sikaru.TraceImportOptionsRequest;
    provider: CreateTraceImportRequest.Provider;
    scope: Sikaru.TraceImportScopeRequest;
    sourceInstance: string;
    tags?: string[];
}

export namespace CreateTraceImportRequest {
    export const Mode = {
            Historical: "historical",
            Selective: "selective"
        } as const;
    export type Mode = typeof Mode[keyof typeof Mode];
    export const Provider = {
            Langsmith: "langsmith",
            Langfuse: "langfuse",
            Braintrust: "braintrust",
            Logfire: "logfire",
            Harbor: "harbor",
            Opentelemetry: "opentelemetry",
            Openinference: "openinference"
        } as const;
    export type Provider = typeof Provider[keyof typeof Provider];
}

File: ruby/lib/Sikaru/trace_imports/types/create_trace_import_request.rb
# frozen_string_literal: true

module Sikaru
  module TraceImports
    module Types
      class CreateTraceImportRequest < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :connection_id, -> { String }, optional: false, nullable: false, api_name: "connectionId"

        field :converter_version, -> { String }, optional: false, nullable: false, api_name: "converterVersion"

        field :dataset, -> { String }, optional: false, nullable: false

        field :external_project_id, -> { String }, optional: false, nullable: false, api_name: "externalProjectId"

        field :grouping, -> { Sikaru::Types::TraceImportGroupingRequest }, optional: true, nullable: false

        field :mode, -> { Sikaru::TraceImports::Types::CreateTraceImportRequestMode }, optional: false, nullable: false

        field :options, -> { Sikaru::Types::TraceImportOptionsRequest }, optional: true, nullable: false

        field :provider, -> { Sikaru::TraceImports::Types::CreateTraceImportRequestProvider }, optional: false, nullable: false

        field :scope, -> { Sikaru::Types::TraceImportScopeRequest }, optional: false, nullable: false

        field :source_instance, -> { String }, optional: false, nullable: false, api_name: "sourceInstance"

        field :tags, -> { Internal::Types::Array[String] }, optional: true, nullable: false
      end
    end
  end
end
